aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/DownloadTask.java

```diff
@@ -114,16 +114,16 @@ public Boolean call() {
             return true;
         } catch (final Exception e) {
             // No need to update the progress listener.
-            if (TransferState.CANCELED.equals(download.state)) {
-                LOGGER.info("Transfer is " + download.state);
+            if (TransferState.PENDING_CANCEL.equals(download.state)) {
+                updater.updateState(download.id, TransferState.CANCELED);
+                LOGGER.info("Transfer is " + TransferState.CANCELED);
                 return false;
             }
 
             // Reset the progress when the transfer is paused.
-            if (TransferState.PAUSED.equals(download.state)) {
-                LOGGER.info("Transfer is " + download.state);
-                ProgressEvent resetEvent = new ProgressEvent(0);
-                resetEvent.setEventCode(ProgressEvent.RESET_EVENT_CODE);
```

Comment:
Are we not resetting on pause on anymore? If so, do we need to update documentation?

Reply:
if you look closely, the lines i removed were noops. Im not sure if the original intent was to pass in a `new ProgressEvent(0)` with reset flag, but this object never gets used and a new progress event without flag is reinstantiated instead. Perhaps a different bug, but I didn't want to introduce yet another behavior change in this PR.